### PR TITLE
feat: default min and max axis range

### PR DESCRIPTION
**Pull request by ZeRego**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #2244 

### Preview:

default min and max range

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/frontend/src/hooks/echarts/useEcharts.test.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        ).toBeUndefined();
+        expect(
+            getAxisDefaultMinValue({
+                min: new Date('2021-03-10T00:00:00.000Z'),
+                max: new Date('2021-03-10T00:00:00.100Z'),
+            }),
+        ).toBeUndefined();
+    });
+
+    test('get axis default min value for positive numbers', () => {
```
Comment (owlas) [inline packages/frontend/src/hooks/echarts/useEcharts.test.ts:22]: Can we split this test into those we expect to be undefined vs. return a minimum value?

File: packages/frontend/src/hooks/echarts/useEcharts.ts [hunk 2]
```diff
@@ -66,6 +66,46 @@ const getAxisTypeFromField = (item?: Field): string => {
...
 
+export const getAxisDefaultMinValue = ({
+    min,
+    max,
+}: {
+    min: any;
+    max: any;
+}) => {
+    if (
+        isNaN(parseInt(min)) ||
```
Comment (owlas) [inline packages/frontend/src/hooks/echarts/useEcharts.ts:77]: why not parse float?

File: packages/frontend/src/hooks/echarts/useEcharts.ts [hunk 3]
```diff
@@ -66,6 +66,46 @@ const getAxisTypeFromField = (item?: Field): string => {
     }
 };
 
+export const getAxisDefaultMinValue = ({
```
Comment (owlas) [inline packages/frontend/src/hooks/echarts/useEcharts.ts:69]: Could these be simplified into a function that takes the existing min, max and returns a new one?

I also wonder if the logic inside could be made more explicit:

```
switch(Math.sign(min) + Math.sign(max)) {
  case: 2
    # both are positive - adjust min value
  case: 0
    # data is both sides of zero - do nothing
  case: -2
    # both are negative - adjust max value

**Review by owlas — CHANGES_REQUESTED** [review]

**Comment by TuringLovesDeathMetal** [comment]
@ZeRego just played around with it a bit. Looks great!

One change (I know I said the opposite earlier, sorry!) but can we make it so that only line charts + scatter plots have the auto range? 

Bar charts should always start from 0 by default (basically, leave the echarts default for barcharts)

File: packages/frontend/src/hooks/echarts/useEcharts.ts (hunk 2 above)
Comment (ZeRego) [reply packages/frontend/src/hooks/echarts/useEcharts.ts:77]: It's only purpose was to handle edge cases with strings. E.g `isNan("")` returns true :/ 
I can remove completely the parseInt and just check for empty strings.

**Review by TuringLovesDeathMetal — APPROVED** [review]
Looks great! 😚 👌

**Pull request merged**